Microsoft.Toolkit.HighPerformance/Enumerables/ReadOnlySpanEnumerable{T}.cs

```diff
@@ -43,7 +43,7 @@ public ReadOnlySpanEnumerable(ReadOnlySpan<T> span)
         /// </summary>
         /// <returns>An <see cref="ReadOnlySpanEnumerable{T}"/> instance targeting the current <see cref="ReadOnlySpan{T}"/> value.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public ReadOnlySpanEnumerable<T> GetEnumerator() => this;
+        public readonly ReadOnlySpanEnumerable<T> GetEnumerator() => this;
```

Comment:
Is this dependent on #3351?

Reply:
Nope, looks like I fixed this here while working on this PR without realizing I had made the same change in #3351 too 😅

Once the other PR is merged I think this should automatically be detected by git anyway so it shouldn't cause issues, would you like me to leave this as is or just revert this change just in case? I mean, this PR will already have a bunch more conflicts anyway so I think this one should be fine to leave for now? 🤔

Reply:
Let's just resolve #3351 first as I think that's ready to go for sure, and then we can ensure git does the safe merge here first when updating this PR.